static/js/admin/feature_form.js

```diff
@@ -86,6 +75,384 @@ function toggleMilestones(status) {
+    INTENT_SHIP:           HIDDEN,
+    INTENT_REMOVE:         HIDDEN
+  },
+  'experiment_risks': {
+    INTENT_IMPLEMENT:      HIDDEN,
+    INTENT_EXPERIMENT:     VISIBLE_REQUIRED,
+    INTENT_EXTEND_TRIAL:   VISIBLE_REQUIRED,
+    INTENT_IMPLEMENT_SHIP: HIDDEN,
+    INTENT_SHIP:           HIDDEN,
+    INTENT_REMOVE:         HIDDEN
+  },
+  'experiment_extension_reason': {
+    INTENT_IMPLEMENT:      HIDDEN,
+    INTENT_EXPERIMENT:     HIDDEN,
+    INTENT_EXTEND_TRIAL:   VISIBLE_REQUIRED,
+    INTENT_IMPLEMENT_SHIP: HIDDEN,
+    INTENT_SHIP:           HIDDEN,
+    INTENT_REMOVE:         HIDDEN
+  },
+  'ongoing_constraints': {
+    INTENT_IMPLEMENT:      HIDDEN,
+    INTENT_EXPERIMENT:     VISIBLE_OPTIONAL,
+    INTENT_EXTEND_TRIAL:   VISIBLE_OPTIONAL,
+    INTENT_IMPLEMENT_SHIP: HIDDEN,
+    INTENT_SHIP:           HIDDEN,
+    INTENT_REMOVE:         HIDDEN
+  },
+  'all_platforms_descr': {
+    INTENT_IMPLEMENT:      VISIBLE_OPTIONAL,
+    INTENT_EXPERIMENT:     HIDDEN,
+    INTENT_EXTEND_TRIAL:   HIDDEN,
+    INTENT_IMPLEMENT_SHIP: VISIBLE_OPTIONAL,
+    INTENT_SHIP:           VISIBLE_OPTIONAL,
+    INTENT_REMOVE:         VISIBLE_OPTIONAL
+  },
+  'ergonomics_risks': {
+    INTENT_IMPLEMENT:      VISIBLE_OPTIONAL,
+    INTENT_EXPERIMENT:     HIDDEN,
+    INTENT_EXTEND_TRIAL:   HIDDEN,
+    INTENT_IMPLEMENT_SHIP: VISIBLE_OPTIONAL,
+    INTENT_SHIP:           VISIBLE_OPTIONAL,
+    INTENT_REMOVE:         VISIBLE_OPTIONAL
+  },
+  'activation_risks': {
+    INTENT_IMPLEMENT:      VISIBLE_OPTIONAL,
+    INTENT_EXPERIMENT:     HIDDEN,
+    INTENT_EXTEND_TRIAL:   HIDDEN,
+    INTENT_IMPLEMENT_SHIP: VISIBLE_OPTIONAL,
+    INTENT_SHIP:           VISIBLE_OPTIONAL,
+    INTENT_REMOVE:         VISIBLE_OPTIONAL
+  },
+  'security_risks': {
+    INTENT_IMPLEMENT:      VISIBLE_OPTIONAL,
+    INTENT_EXPERIMENT:     HIDDEN,
+    INTENT_EXTEND_TRIAL:   HIDDEN,
+    INTENT_IMPLEMENT_SHIP: VISIBLE_OPTIONAL,
+    INTENT_SHIP:           VISIBLE_OPTIONAL,
+    INTENT_REMOVE:         VISIBLE_OPTIONAL
+  },
+  'origin_trial_feedback_url': {
+    INTENT_IMPLEMENT:      HIDDEN,
+    INTENT_EXPERIMENT:     HIDDEN,
+    INTENT_EXTEND_TRIAL:   HIDDEN,
+    INTENT_IMPLEMENT_SHIP: HIDDEN,
+    INTENT_SHIP:           VISIBLE_OPTIONAL,
+    INTENT_REMOVE:         VISIBLE_OPTIONAL
+  },
+}
+
+function intentStageChanged(stage) {
+  function set_required(id, required) {
+    if (required) {
+      document.querySelector('#id_' + id).setAttribute('required', 'required');
+    } else {
+      document.querySelector('#id_' + id).removeAttribute('required');
+    }
+  }
+
+  function show(id) {
+    // Clear the inline style for the control's table row.
+    document.querySelector('#id_' + id).parentNode.parentNode.style = "";
+  }
+
+  function hide(id) {
+    // Set the control's table row style to display: none.
+    document.querySelector('#id_' + id).parentNode.parentNode
+	.style = "display: none;";
+  }
+
+  let stage_index = INTENT_IDENTIFIER_NAMES[Number(stage.value)];
```

Comment:
vars should be camelcase

```suggestion
  let stageIndex = INTENT_IDENTIFIER_NAMES[Number(stage.value)];
```